### PR TITLE
feat: 🎸 status component

### PR DESCRIPTION
**Pull request by Exayy**
## Summary

Add a "Status" component which will replace Status Indicator

- It will replace `StatusIndicator` but if we let them both, we simplify our migration
- `babel-plugin-named-exports-order` allow exported story to be in the same order than in file, avoiding strange behavior with `Playground` story

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -14,5 +14,6 @@
         "labelFormat": "[filename]--[local]"
       }
     ]
-  ]
+  ],
+  "plugins": ["babel-plugin-named-exports-order"]
```
Comment (chambo-e) [inline babel.config.json:18]: Can we put this only in storybook config ?

File: src/components/Status/__tests__/index.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import Status, { statusVariants } from '..'
+import { shouldMatchEmotionSnapshot } from '../../../helpers/jestHelpers'
+
+describe('Status', () => {
+  statusVariants.forEach(variant => {
+    test(`render ${variant}`, () =>
```
Comment (chambo-e) [inline src/components/Status/__tests__/index.tsx:7]: test.each maybe ?

File: src/components/Status/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    `,
+  neutral: ({ theme }: { theme: Theme }) => `
+      background-color: ${theme.colors.neutral.backgroundStronger};
+    `,
+  success: ({ theme }: { theme: Theme }) => `
+      background-color: ${theme.colors.success.backgroundStrong};
+    `,
+  warning: ({ theme }: { theme: Theme }) => `
+      background-color: ${theme.colors.warning.backgroundStrong};
+      color: ${theme.colors.primary.text};
```
Comment (chambo-e) [inline src/components/Status/index.tsx:22]: color ?

**Review by chambo-e — COMMENTED** [review]
Shouldn't we delete StatusIndicator too?

File: src/components/Status/index.tsx (hunk 3 above)
Comment (Exayy) [reply src/components/Status/index.tsx]: done

File: src/components/Status/__tests__/index.tsx (hunk 2 above)
Comment (Exayy) [reply src/components/Status/__tests__/index.tsx]: done 👌

File: babel.config.json (hunk 1 above)
Comment (Exayy) [reply babel.config.json]: done

**Comment by Exayy** [comment]
> Shouldn't we delete StatusIndicator too?

We should probably let this component until migration isn't done

File: package.json [hunk 4]
```diff
@@ -185,6 +185,7 @@
     "@types/intl-tel-input": "17.0.4",
     "@types/react-datepicker": "4.4.1",
     "@xstyled/emotion": "2.5.0",
+    "babel-plugin-named-exports-order": "^0.0.2",
```
Comment (chambo-e) [inline package.json:188]: devDeps maybe ?

Comment (QuiiBz) [reply package.json:188]: We can also move the above `@types` packages to devdeps

**Review by QuiiBz — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (chambo-e) [reply package.json:188]: Nope, @types/intl-tel-input and @types/react-datepicker are exported in the definition bundle :)

Comment (Exayy) [reply package.json:188]: Done

File: src/components/Status/index.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  className,
+  tooltip,
+  variant,
+}: StatusProps): JSX.Element => (
+  <Tooltip text={tooltip}>
+    <StyledCircle
+      className={className}
+      css={animated && cssAnimation}
+      variant={variant}
+    />
```
Comment (matthprost) [inline src/components/Status/index.tsx:60]: Can't we pass animated prop directly to `StyledCircle` and condition with ternary `animation: ${flash} linear 1s infinite;` there directly ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Status/index.tsx (hunk 5 above)
Comment (Exayy) [reply src/components/Status/index.tsx]: done

**Review by matthprost — APPROVED** [review]

**Pull request merged**